lib/write.js

```diff
@@ -71,42 +73,30 @@ var Write = Juttle.proc.sink.extend({
         });
 
         return points;
-    },
-
-    insertPoints: function(points) {
-        var self = this;
+    }
 
+    insertPoints(points) {
         if (points.length === 0) { return; }
 
-        this.inserts_in_progress++;
-
         var knex = db.getDbConnection();
 
-        return knex(this.table).insert(points)
-        .catch(function(err) {
-            self.trigger('error', self.runtime_error('RT-INTERNAL-ERROR', {
+        return knex(this.table).insert(points.splice(0,1000))
+        .catch((err) => {
+            this.trigger('error', new errors.runtimeError('RT-INTERNAL-ERROR', {
                  error: err.toString()
              }));
-        }).finally(function() {
-            self.tryFinish(true);
+        })
+        .then(() => {
+            if (points.length) {
```

Comment:
on line 79 points could be an array of length 200. On 90 it could be 0. So we need both checks.